llvm/include/llvm/Target/TargetSelectionDAG.td

```diff
@@ -818,6 +821,10 @@ def step_vector : SDNode<"ISD::STEP_VECTOR", SDTypeProfile<1, 1,
 def scalar_to_vector : SDNode<"ISD::SCALAR_TO_VECTOR", SDTypeProfile<1, 1, []>,
                               []>;
 
+def build_pair : SDNode<"ISD::BUILD_PAIR", SDTypeProfile<1, 2,
```

Comment:
https://github.com/llvm/llvm-project/blob/cf5947be13e7af67219379e07bc0128f1f1e7f88/llvm/include/llvm/CodeGen/ISDOpcodes.h#L233-L236

BUILD_PAIR should not survive past legalization.

Reply:
Here I'm using it to replace `BUILD_VECTOR` as I'll be lowering it to a packing `mov.b64` later. It's very useful compared to defining a new opcode in NVPTXISD.